app/models/item.rb

```diff
@@ -59,4 +67,45 @@ def perform_associated_job
       ExtensionJob.perform_later(url: content['url'], user_id: user_uuid, extension_id: uuid)
     end
   end
+
+  private
+
+  def cleanup_revisions_for_a_day(days_from_today, allowed_revisions_count)
+    date = Time.now.utc.to_date - days_from_today
+    revisions_from_date = revisions.where(created_at: date.midnight..date.end_of_day).pluck(:uuid)
+
+    if revisions_from_date.length > allowed_revisions_count
+      keep_every_nth = (revisions_from_date.length.to_f / allowed_revisions_count).ceil
+      revisions_to_keep = []
+      (0..revisions_from_date.length - keep_every_nth).step(keep_every_nth) do |n|
+        revisions_to_keep.push(revisions_from_date[n])
+      end
+
+      Revision
+        .where(created_at: date.midnight..date.end_of_day)
+        .where.not(uuid: revisions_to_keep)
+        .destroy_all
+    end
+  end
+
+  def cleanup_excessive_revisions(days = User::REVISIONS_RETENTION_DAYS)
+    days.times do |days_from_today|
+      allowed_revisions_count = [days - days_from_today + MIN_REVISIONS_PER_DAY, MAX_REVISIONS_PER_DAY].min
```

Comment:
I'm not sure I'm following this. So here's how the current ext works:

Today (0 days from present): 30 revisions
Today - 1 (yesterday) (1 days from present): 29 revisions
Today - 2 (2 days from present): 28 revisions
...
Today - 40 (40 days from present): min_revisions : 2

Whereas with this current functionality, it seems we have:

Today (0 days from present): `[30 - 0 + MIN_REVISIONS_PER_DAY, MAX_REVISIONS_PER_DAY].min = [34, 8].min = 8`
Today - 1 (yesterday) (1 days from present):  `[30 - 1 + MIN_REVISIONS_PER_DAY, MAX_REVISIONS_PER_DAY].min = [33, 8].min = 8`
...

So it always returns 8?

In any case, we should also add a test that correctly anticipates the decaying revision count.

Reply:
Isn't this the same as in the extension?

https://github.com/standardnotes/note-history-ext/blob/efdb5db80ab04a1fd907d6f9be6c803a93ad3519/app/models/revisions/item.rb#L29